### ISSUE TITLE
Unable to escalate privileges if Fingerprints are configured in Gnome

### ISSUE THREAD

**Issue by barnscott** [cockpit-project/cockpit#19479]
### Explain what happens

TLDR
I am unable to authenticate and access "Administrator access" if Fingerprints are configured for my user in Gnome.

Package source: Flatpak - Clockpit-client v301 on Silverblue-38

INFO
I downloaded Cockpit-Client (flatpak). I logged into Cockpit via localhost and I am using my regular user which has admin permission. However, I am unable to access "Administrator access" while Fingerprints are configured in Gnome: when i select "Limited access" I am not prompted to authenticate. Log from journalctl are available below while Fingerprints were enabled.

After deleting Fingerprints for my user in Gnome-settings, and closing/opening Cockpit, administrator access works as expected in Gnome.

### Version of Cockpit

301

### Where is the problem in Cockpit?

Accounts

### Server operating system

Fedora

### Server operating system version

fedora-silverblue-38

### What browsers are you using?

Firefox

### System log

```shell
Oct 13 10:00:11 laptop-l gnome-character[65685]: JS LOG: Characters Application exiting
Oct 13 10:00:13 laptop-l systemd[2053]: Started app-gnome-org.cockpit_project.CockpitClient-66643.scope - Application launched by gnome-shell.
Oct 13 10:00:13 laptop-l systemd[2053]: Started app-flatpak-org.cockpit_project.CockpitClient-66643.scope.
Oct 13 10:00:14 laptop-l systemd[2053]: Started app-flatpak-org.cockpit_project.CockpitClient-66721.scope.
Oct 13 10:00:14 laptop-l systemd[2053]: Started app-flatpak-org.cockpit_project.CockpitClient-66757.scope.
Oct 13 10:00:14 laptop-l org.cockpit_project.CockpitClient.desktop[66808]: Failed to create /var/home/barnabas/.var/app/org.cockpit_project.CockpitClient/cache for shader cache (No such file or directory)---disabling.
Oct 13 10:00:14 laptop-l rtkit-daemon[1068]: Successfully made thread 66821 of process 66808 (/usr/libexec/webkit2gtk-4.1/WebKitWebProcess) owned by '1000' RT at priority 5.
Oct 13 10:00:18 laptop-l audit: BPF prog-id=503 op=LOAD
Oct 13 10:00:18 laptop-l audit: BPF prog-id=504 op=LOAD
Oct 13 10:00:18 laptop-l audit: BPF prog-id=505 op=LOAD
Oct 13 10:00:18 laptop-l systemd[1]: Starting systemd-hostnamed.service - Hostname Service...
Oct 13 10:00:18 laptop-l systemd[1]: Started systemd-hostnamed.service - Hostname Service.
Oct 13 10:00:18 laptop-l audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 13 10:00:19 laptop-l audit: BPF prog-id=506 op=LOAD
Oct 13 10:00:19 laptop-l audit: BPF prog-id=507 op=LOAD
Oct 13 10:00:19 laptop-l audit: BPF prog-id=508 op=LOAD
Oct 13 10:00:19 laptop-l systemd[1]: Starting systemd-timedated.service - Time & Date Service...
Oct 13 10:00:19 laptop-l realmd[2137]: client using service: :1.4717
Oct 13 10:00:19 laptop-l realmd[2137]: holding daemon: :1.4717
Oct 13 10:00:19 laptop-l systemd[1]: Started systemd-timedated.service - Time & Date Service.
Oct 13 10:00:19 laptop-l audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-timedated comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 13 10:00:22 laptop-l audit: BPF prog-id=509 op=LOAD
Oct 13 10:00:22 laptop-l systemd[1]: Starting fprintd.service - Fingerprint Authentication Daemon...
Oct 13 10:00:22 laptop-l systemd[1]: Started fprintd.service - Fingerprint Authentication Daemon.
Oct 13 10:00:22 laptop-l audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=fprintd comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 13 10:00:49 laptop-l systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Oct 13 10:00:49 laptop-l audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 13 10:00:49 laptop-l audit: BPF prog-id=505 op=UNLOAD
Oct 13 10:00:49 laptop-l audit: BPF prog-id=504 op=UNLOAD
Oct 13 10:00:49 laptop-l audit: BPF prog-id=503 op=UNLOAD
Oct 13 10:00:49 laptop-l systemd[1]: systemd-timedated.service: Deactivated successfully.
Oct 13 10:00:49 laptop-l audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-timedated comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 13 10:00:49 laptop-l audit: BPF prog-id=508 op=UNLOAD
Oct 13 10:00:49 laptop-l audit: BPF prog-id=507 op=UNLOAD
Oct 13 10:00:49 laptop-l audit: BPF prog-id=506 op=UNLOAD
Oct 13 10:00:52 laptop-l sudo[67375]: pam_unix(sudo:auth): conversation failed
Oct 13 10:00:52 laptop-l audit[67375]: USER_AUTH pid=67375 uid=1000 auid=1000 ses=3 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='op=PAM:authentication grantors=? acct="barnabas" exe="/usr/bin/sudo" hostname=? addr=? terminal=? res=failed'
Oct 13 10:00:52 laptop-l sudo[67375]: pam_unix(sudo:auth): auth could not identify password for [barnabas]
Oct 13 10:00:53 laptop-l audit[67375]: USER_CMD pid=67375 uid=1000 auid=1000 ses=3 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='cwd="/var/home/barnabas" cmd=707974686F6E33202D6963202320636F636B7069742D627269646765202D2D70726976696C65676564 exe="/usr/bin/sudo" terminal=? res=failed'
```

**Comment by jelly** [comment]
Seeing the SELinux warnings, makes me wonder if that's the issue. Cockpit tries to auth via PAM which from a Cockpit context can't access the fingerprint reader?

**Comment by martinpitt** [comment]
While cockpit runs under SELinux in a normal rpm installation, it shouldn't do that through the flatpak -- but of course SELinux is always good for a surprise. So trying this with `sudo setenforce 0` and comparing is certainly useful.
